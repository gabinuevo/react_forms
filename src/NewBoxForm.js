import React, { Component } from 'react';

/**
 *  render a form that when submitted, creates a new Box. 
 *  input: {
 *          background color, 
 *          width 
 *          height 
 *            }
 *  When the form is submitted, clear the input values.
 */

class NewBoxForm extends Component {

    constructor(props) {
        super(props);
        this.state = {
            height: '',
            width: '',
            backgroundColor: ''
            };
        this.handleChange = this.handleChange.bing(this);
        this.handleSubmit = this.handleSubmit.bing(this);
    } 

    handleChange() {

    }

    handleSubmit() {

    }

    render() {
        return (
            <form onSubmit={this.handleSubmit}>
                
                <label for="width"> Width: </label>
                <input name="width" value={this.state.width} onChange={this.handleChange}/>
                
                <label for="height"> Height: </label>
                <input name="height" value={this.state.height} onChange={this.handleChange}/>
                
                <label for="backgroundColor"> Background Color: </label>
                <input name="backgroundColor" value={this.state.backgroundColor} onChange={this.handleChange}/>

                <button>  Add New Box!  </button>
            </form>
        );
    }
}

export default NewBoxForm;