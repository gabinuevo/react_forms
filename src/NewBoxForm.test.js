import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import NewBoxForm from "./NewBoxForm";

let wrapper;

it("renders succesfully", function() {
    shallow(<NewBoxForm />);
});

it("matches snapshot on initial render", function() {
    wrapper = shallow(<NewBoxForm />);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });



it("tests our form inputs: width, height, backgroundColor, which should update state", function() {

    let wrapper = mount(<NewBoxForm />);
    const height = wrapper.find("#height")
    height.instance().value = "100px"
    height.simulate("change")

    expect(wrapper.state().height).toEqual("100px");

    const width = wrapper.find("#width");
    width.instance().value = '200px';
    width.simulate("change");

    expect(wrapper.state().width).toEqual('200px');

    const backgroundColor = wrapper.find("#backgroundColor");
    backgroundColor.instance().value = 'black';
    backgroundColor.simulate("change");

    expect(wrapper.state().backgroundColor).toEqual('black');

});  


it("runs a mocked fn on submit", function () {
    const submitFn = jest.fn();
    let wrapper = mount(
      <NewBoxForm triggerAddNewBox={submitFn} />
    );
    const form = wrapper.find("form")
  
    form.simulate("submit")
  
    expect(submitFn).toHaveBeenCalled();
});


it("resets state on submit", function () {
  const submitFn = jest.fn();
  
  let wrapper = mount(<NewBoxForm triggerAddNewBox={submitFn} />);
  
  const height = wrapper.find("#height")
  height.instance().value = "100px"
  height.simulate("change")

  expect(wrapper.state().height).toEqual("100px");

  const width = wrapper.find("#width");
  width.instance().value = "200px";
  width.simulate("change");

  expect(wrapper.state().width).toEqual("200px");
  
  const backgroundColor = wrapper.find("#backgroundColor")
  backgroundColor.instance().value = "red"
  backgroundColor.simulate("change")

  expect(wrapper.state().backgroundColor).toEqual("red");

  const form = wrapper.find("form");
  form.simulate("submit");

  // after submit, we expect the state to reset
  expect(wrapper.state()).toEqual({
    height: '',
    width: '',
    backgroundColor: ''
  })
});
