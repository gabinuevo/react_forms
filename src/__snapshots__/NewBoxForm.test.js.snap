// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot on initial render 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="height"
  >
     Height: 
  </label>
  <input
    id="height"
    name="height"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="width"
  >
     Width: 
  </label>
  <input
    id="width"
    name="width"
    onChange={[Function]}
    value=""
  />
  <label
    htmlFor="backgroundColor"
  >
     Background Color: 
  </label>
  <input
    id="backgroundColor"
    name="backgroundColor"
    onChange={[Function]}
    value=""
  />
  <button>
      Add New Box!  
  </button>
</form>
`;
